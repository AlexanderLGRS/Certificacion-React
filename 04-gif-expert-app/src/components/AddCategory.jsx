import { useState } from "react"

export function AddCategory(props) {
    const [inputValue, setInputValue] = useState('')

    const inputChangeHandler = (event) => {
        setInputValue(event.target.value)
    }
    const submitHandler = (event) => {
        event.preventDefault()
        if(inputValue === '')return
        props.onSubmitHandler(inputValue)
        setInputValue('')
    }
    return (
        <form onSubmit={submitHandler}>
            <input
                type="text"
                placeholder="Buscar gifs"
                value={inputValue}
                onChange={inputChangeHandler}
                autoFocus
            />
        </form>
    )
}