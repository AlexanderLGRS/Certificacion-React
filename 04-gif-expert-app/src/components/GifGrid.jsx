import PropTypes from "prop-types";
import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifItem } from "./";

export function GifGrid({ category }) {

    const { images, isLoading } = useFetchGifs(category)

    return (
        <>
            <h3>{category}</h3>
            {isLoading && <h1>Cargando...</h1>}
            <div className="card-grid">
                {images.map((gif) => (
                    <GifItem
                        key={gif.id}
                        {...gif}
                    />
                ))}
            </div>
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}