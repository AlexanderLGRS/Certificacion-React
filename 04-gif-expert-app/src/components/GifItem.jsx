import PropTypes from 'prop-types'
export function GifItem({ id, url, title }) {
    return (
        <div className="card" key={id}>
            <img src={url} alt={title} />
            <p>{title}</p>
        </div>
    )
}
GifItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
    id: PropTypes.string,
}