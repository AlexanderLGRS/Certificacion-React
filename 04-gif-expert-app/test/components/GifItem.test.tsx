import { render, screen } from '@testing-library/react';
import { GifItem } from '../../src/components/GifItem';

describe('Pruebas en GifItem', () => {
  const titulo = 'Hola soy Alex';
  const url = 'http://soy_la_url/';
  test('Debe hacer match con el snapshot', () => {
    const { container } = render(<GifItem title={titulo} url={url} />);
    expect(container).toMatchSnapshot();
  });
  test('Debe renderizar la imagen con el alt y el src correctos', () => {
    render(<GifItem title={titulo} url={url} />);
    const {src, alt} = screen.getByRole('img')
    expect(src).toBe(url)
    expect(alt).toBe(titulo)
  });
  test('Debe renderizar el titulo', () => {
    render(<GifItem title={titulo} url={url} />);
    expect(screen.getByText(titulo)).toBeTruthy();
  });
});
