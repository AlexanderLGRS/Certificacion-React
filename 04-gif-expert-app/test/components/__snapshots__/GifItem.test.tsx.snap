// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifItem Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Hola soy Alex"
      src="http://soy_la_url/"
    />
    <p>
      Hola soy Alex
    </p>
  </div>
</div>
`;
